opts.go

```diff
@@ -66,3 +69,17 @@ func WithSnapshot(memFilePath, snapshotPath string, opts ...WithSnapshotOpt) Opt
 		m.Handlers.FcInit = loadSnapshotHandlerList
 	}
 }
+
+// MemoryBackendType sets the memory backend type to the given value.
+func MemoryBackendType(typ string) WithSnapshotOpt {
+	return func(cfg *SnapshotConfig) {
+		if cfg.MemBackend == nil {
+			cfg.MemBackend = &models.MemoryBackend{
+				BackendPath: String(cfg.MemFilePath),
+			}
+		}
+		cfg.MemBackend.BackendType = String(typ)
+		// Clear MemFilePath since only one of MemFilePath or MemBackend is allowed.
```

Comment:
The comment here needs to be more explicit `MemFilePath is ignored if MemBackend is set` and